### PR TITLE
paths.rs: Add a test suite for iterators

### PR DESCRIPTION
**Pull request by federicomenaquintero**
This is the test suite I mentioned in https://github.com/gtk-rs/cairo/issues/87, now with the fixed Path::iter().

### CODE REVIEW

File: src/paths.rs [hunk 1]
```diff
@@ -105,3 +106,111 @@ impl Iterator for PathSegments {
...
+        let path = cr.copy_path();
+
+        assert_path_equals_segments(&path,
+                                    &vec![PathSegment::MoveTo((1.0, 2.0)),
+                                          PathSegment::CurveTo((3.0, 4.0), (5.0, 6.0), (7.0, 8.0)),
+                                          PathSegment::ClosePath,
+                                          PathSegment::MoveTo((1.0, 2.0)),
+                                          PathSegment::LineTo((9.0, 10.0))]);
+    }
+
```
Comment (GuillaumeGomez) [inline src/paths.rs]: Extra empty line.

File: src/paths.rs [hunk 2]
```diff
@@ -105,3 +106,111 @@ impl Iterator for PathSegments {
...
+        // Note that Cairo represents a close_path as closepath+moveto,
+        // so that the next subpath will have a starting point,
+        // from the extra moveto.
+        assert_path_equals_segments(&path,
+                                    &vec![PathSegment::MoveTo((1.0, 2.0)),
+                                          PathSegment::ClosePath,
+                                          PathSegment::MoveTo((1.0, 2.0))]);
+    }
+    #[test]
```
Comment (GuillaumeGomez) [inline src/paths.rs]: Missing empty line.

**Comment by federicomenaquintero** [comment]
Fixed the nits.  Thanks for the careful eye.

**Comment by federicomenaquintero** [comment]
This pull request already got merged; closing.

**Pull request closed without merging**